runtime/src/main/java/org/corfudb/runtime/collections/streaming/StreamingTask.java

```diff
@@ -59,41 +62,60 @@
 
     private final ExecutorService workerPool;
 
-    private final DeltaStream stream;
-
     private final AtomicReference<StreamStatus> status;
 
     private volatile Throwable error;
 
-    public StreamingTask(CorfuRuntime runtime, ExecutorService workerPool, String namespace, String streamTag,
+    private final List<DeltaStream> streamsList;
+
+    private DeltaStream currentStream = null;
+
+    public StreamingTask(CorfuRuntime runtime, ExecutorService workerPool, String appNamespace, String appStreamTag,
```

Comment:
I would suggest having a single constructor which takes the nsToTableNames map and list of stream tags instead of having 2 separate ones.